(maint) Standardize params vs. options and use symbols

nicklewis opened the pull request:
Previously, we split out params and options into separate hashes
based on whether the keys started with _, but then left the keys that
way. That meant that the internals of Bolt were littered with code that
included leading underscores and thus had some evidence of where those
options originated from. As the leading underscore is an implementation
detail of the Puppet functions, we now consistently strip them out when
exiting the Puppet language and standardize them to symbols.

nicklewis commented on spec/bolt/transport/local_spec.rb:
```diff
@@ -131,7 +131,7 @@
 
     context 'with slow input' do
-      let(:file_size) { 100 }
+      let(:file_size) { 10 }
```
This is an unrelated change but it cuts this test from 20 seconds to 2 seconds and I don't _think_ it has an impact on the behavior of the test. I couldn't quite tell what problem it was meant to spot though, so I can't be sure.

donoghuc replied:
This seems totally reasonable.

puppetcla commented:
CLA signed by all contributors.

adreyer commented:
Note how plugins use options.

https://github.com/puppetlabs/bolt/blob/master/lib/bolt/plugin/module.rb#L167

donoghuc commented on bolt-modules/boltlib/lib/puppet/functions/upload_file.rb:
```diff
@@ -47,18 +47,19 @@
 
-    options ||= {}
-    options = options.merge('_description' => description) if description
+    options = options.select { |opt| opt.start_with?('_') }.map { |k, v| [k.gsub(/^_/, '').to_sym, v] }.to_h
```
Why do you use `gsub` here? Looks like for `run_plan` and this you use it, but for the others its just `sub`? Are there multiple underscores to replace for only these two?

nicklewis replied:
I think I was just inconsistent. I'll standardize them all the `sub`.

nicklewis commented:
Updated to use `sub` everywhere and rebased on master. I think the noop thing is sort of related to #1207 so might be better to address there. They're both about the fact that we have half-implemented noop support for plans.
